src/main/antlr/GroovyParser.g4

```diff
@@ -255,15 +255,15 @@ locals[ int t ]
 
 // t    see the comment of classDeclaration
 classBody[int t]
-    :   LBRACE nls
+    :   LBRACE
         (
             /* Only enum can have enum constants */
             { 2 == $t }?
-            enumConstants? sep?
+            nls enumConstants
         |
-
         )
-        classBodyDeclaration[$t]? (sep classBodyDeclaration[$t])* sep? RBRACE
+        sep?
+        (classBodyDeclaration[$t] (sep classBodyDeclaration[$t])*)? sep? RBRACE
```

Comment:
📝 This change falls out from the following in the original rule:

```
classBodyDeclaration[$t]? (sep classBodyDeclaration[$t])*
```

While it was perhaps not intentional, one interpretation of this is:

```
(sep classBodyDeclaration[$t])+
```

Which means `sep` can appear before the first `classBodyDeclaration` for enum types or any other type. This allowed me to extract the `sep?` from the enum-specific alternative, and leverage that for the `nls` following `LBRACE` in non-enum types.